src/RenderView.tsx

```diff
@@ -66,6 +74,59 @@ export default function RenderView(props: Props) {
 		return result;
 	});
 
+	const [cameraControls] = useState(() => new PointerLockControls(camera, renderer.domElement));
+
+	/* Use camera controls */
+	useEffect(() => {
+		const moveMeters = 0.05;
+		/* Only allow flight while mouse is in visualization area */
+		renderer.domElement.addEventListener('mouseenter', () => document.addEventListener("keydown", keyPressHandler, false), false);
```

Comment:
if you add event listeners in useEffect you should remove them in the cleanup function

Reply:
Done. I'm still getting the hang of React. That makes sense.